### DISCUSSION TITLE
A merge commit message written in the GitHub UI can be lost/dropped

### DISCUSSION THREAD

**Discussion by brekelj1** [community/community#39881] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

**Situation**
When a user wants to merge a pull request via the user interface, the user can edit the commit message in the UI.
If the pull request is not yet mergeable and auto-merge is enabled, the button shows up as (paraphrasing) "enable automerge".
When the pull request becomes mergeable, the button changes to (paraphrasing) "merge now".

**Feedback**
The problem occurs when in between clicking "enable automerge", and the user editing and confirming the commit message, the pull request becomes mergeable. In this case, when the user tries to confirm the commit message and go ahead with enabling automerge, an error occurs and the commit message text is lost. The user then has to refresh the page and re-enter the commit message after clicking "merge now". This is particularly annoying when the user has spent 5 or more minutes writing a nice commit message.

**Comment by futuresea-dev** [comment]
YEAH

Pull requests let you tell others about changes you've pushed to a branch in a repository on GitHub. Once a pull request is opened, you can discuss and review the potential changes with collaborators and add follow-up commits before your changes are merged into the base branch.

Pull requests are built on the principles of code review and team collaboration. Developers can request reviews from their colleagues and track the build status of their work via PRs. Additionally, using pull requests in the code review process enables you as the leader to track your team's review process as a whole

**Reply by brekelj1** [reply]
It wasn't so much a question, but more very specific feedback around a scenario where GitHub drops/loses the commit msg.
